sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -33,6 +33,85 @@ import org.apache.spark.sql.catalyst.util.{ArrayBasedMapData, GenericArrayData}
 import org.apache.spark.sql.types._
 
 /**
+ * Common base class for [[StaticInvoke]], [[Invoke]], and [[NewInstance]].
+ */
+trait InvokeLike extends Expression with NonSQLExpression {
+
+  def arguments: Seq[Expression]
+
+  def propagateNull: Boolean
+
+  protected lazy val needNullCheck: Boolean = propagateNull && arguments.exists(_.nullable)
+
+  /**
+   * Prepares codes for arguments.
+   *
+   * - generate codes for argument.
+   * - use ctx.splitExpressions() to not exceed 64kb JVM limit while preparing arguments.
+   * - avoid some of nullabilty checking which are not needed because the expression is not
+   *   nullable.
+   * - when needNullCheck == true, short circuit if we found one of arguments is null because
+   *   preparing rest of arguments can be skipped in the case.
+   *
+   * @param ctx a [[CodegenContext]]
+   * @param ev an [[ExprCode]] with unique terms.
+   * @return (code to prepare arguments, argument string, code to set isNull)
+   */
+  def prepareArguments(ctx: CodegenContext, ev: ExprCode): (String, String, String) = {
+
+    val containsNullInArguments = if (needNullCheck) {
+      val containsNullInArguments = ctx.freshName("containsNullInArguments")
+      ctx.addMutableState("boolean", containsNullInArguments, "")
+      containsNullInArguments
+    } else {
+      ""
+    }
+    val argValues = arguments.zipWithIndex.map { case (e, i) =>
+      val argValue = ctx.freshName("argValue")
+      ctx.addMutableState(ctx.javaType(e.dataType), argValue, "")
+      argValue
+    }
+
+    val argCodes = if (needNullCheck) {
+      val reset = s"$containsNullInArguments = false;"
+      val argCodes = arguments.zipWithIndex.map { case (e, i) =>
+        val expr = e.genCode(ctx)
+        val updateContainsNull = if (e.nullable) {
+          s"$containsNullInArguments = ${expr.isNull};"
+        } else {
+          ""
+        }
+        s"""
+          if (!$containsNullInArguments) {
```

Comment:
if `containsNullInArguments == "true"`, we do not need generate this code block. Am I correct?